File: 42_ScientificTesting/file_writer/test_file_writer_soln.py
import file_writer


class MockFM:
    def __init__(self):
        self.one = None
        self.two = []
        self.three = None

    def save_1(self, value):
        self.one = value

    def save_2(self, values):
        self.two = values

    def save_3(self, value):
        self.three = value


def test_compute_value_2_draws_size():
    dp = file_writer.DataProcessor()
    dp.compute_value_2(7)

    # One solution: mock the file manager
    # Could also consider making the members accessible for testing
    # or reading the files from disk for the real version, plus cleanup
    mock_fm = MockFM()
    dp.write_files(mock_fm)
    assert len(mock_fm.two) == 7

File: 42_ScientificTesting/file_writer/test_file_writer_soln.cpp
/*

CE LINK: https://godbolt.org/z/8n8hPo9b3

*/

#include <gtest/gtest.h>

#include "file_writer.hpp"

struct MockFIM {
    double one;
    std::vector<std::string> two;
    int three;

    void save_1(double v){ one = v;}
    void save_2(std::vector<std::string>& v){ two = v; }
    void save_3(int v){ three = v; }
};

TEST(DataProcessor, Compute2DrawsSize) {
    DataProcessor dp{};
    dp.compute_value_2(7);

    // One solution: mock the file manager
    // Could also consider making the members accessible for testing
    // or reading the files from disk for the real version, plus cleanup
    MockFIM file_manager{};
    dp.write_files(file_manager);
    EXPECT_EQ(file_manager.two.size(), 7);
 }



















int main(int argc, char** argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}
